notification-service/src/main/java/com/redhat/parodos/notification/controller/advice/ApiValidationError.java

```diff
@@ -0,0 +1,39 @@
+package com.redhat.parodos.notification.controller.advice;
+
+import java.beans.Introspector;
+import java.util.Locale;
+
+import javax.validation.ConstraintViolation;
+
+import lombok.*;
+
+import org.springframework.context.MessageSource;
+
+@Getter
+@ToString
+public class ApiValidationError {
+
+	String defaultMessage;
+
+	String objectName;
+
+	String field;
+
+	Object rejectedValue;
+
+	String code;
+
+	public static ApiValidationError from(ConstraintViolation<?> violation, MessageSource msgSrc, Locale locale) {
+		ApiValidationError result = new ApiValidationError();
+		result.defaultMessage = msgSrc.getMessage(violation.getMessageTemplate(),
+				new Object[] { violation.getLeafBean().getClass().getSimpleName(),
+						violation.getPropertyPath().toString(), violation.getInvalidValue() },
+				violation.getMessage(), locale);
+		result.objectName = Introspector.decapitalize(violation.getRootBean().getClass().getSimpleName());
+		result.field = String.valueOf(violation.getPropertyPath());
+		result.rejectedValue = violation.getInvalidValue();
+		result.code = violation.getMessageTemplate();
+		return result;
```

Comment:
maybe we can set this class as singleton and **return this** in this static method?

Reply:
+1
On the one hand, this is practically a customized builder for this class.
However, conventions...conventions... :-)

Hence, my suggestion is decouple the dto from the instance creation either as Richard has suggested or by introducing a Utils class for it or `new ApiValidationErrorMapper().map()`

Reply:
Just realized we have WorkFlowDefinitionResponseDTO.fromEntity, so the convention isn't enforced yet.

Reply:
I don't think it should be singleton here, as we might need more than one instance during an execution

Reply:
Changed it to builder, the static function use builder to create the instance